client/checkout/woopay/express-button/woopay-express-checkout-button.js

```diff
@@ -46,12 +47,23 @@ export const WoopayExpressCheckoutButton = ( {
 	const onClickCallbackRef = useRef( null );
 	const buttonRef = useRef( null );
 	const isLoadingRef = useRef( false );
-	const { type: buttonType, height, size, theme, context } = buttonSettings;
+	let {
+		type: buttonType,
+		height: buttonHeight,
+		theme,
+		context,
+	} = buttonSettings;
 	const [ isLoading, setIsLoading ] = useState( false );
 	const [ buttonWidthType, setButtonWidthType ] = useState(
 		buttonWidthTypes.wide
 	);
 
+	// If we are on the checkout block, we receive button attributes which overwrite the extension specific settings
+	if ( buttonAttributes !== undefined ) {
```

Comment:
This type check is a little safer

```suggestion
	if ( typeof buttonAttributes !== 'undefined' ) {
```